auth-table-based-service/src/main/java/io/stargate/auth/table/AuthzTableBasedService.java

```diff
@@ -40,6 +42,29 @@ public ResultSet authorizedDataRead(
     // Cannot perform authorization with a table based token so just return
     return action.call();
   }
+  /**
+   * Authorization for data access is not provided by table based tokens so all authorization will
+   * be deferred to the underlying permissions assigned to the role the token maps to.
+   *
+   * <p>{@inheritdoc}
+   */
+  @Override
+  public CompletionStage<ResultSet> authorizedAsyncDataRead(
+      Callable<CompletionStage<ResultSet>> action,
+      String token,
+      String keyspace,
+      String table,
+      List<TypedKeyValue> typedKeyValues,
+      SourceAPI sourceAPI) {
+    // Cannot perform authorization with a table based token so just wrap and return
+    try {
+      return action.call();
+    } catch (Exception e) {
+      CompletableFuture<ResultSet> failedFuture = new CompletableFuture<>();
+      failedFuture.completeExceptionally(e);
+      return failedFuture;
+    }
```

Comment:
Suggestion: If we used an functional interface without checked exception for `action` (e.g. `Supplier`), we could write the body as:
```
    return CompletableFuture
        .completedFuture(null)
        .thenCompose(__ -> action.get());
```

The benefit is that it's more async-native and handles edge cases (like `Error`) gracefully.

WDYT?

Reply:
Yeah my choice would have been `Supplier` if I wrote it from scratch, but the sync version already uses `Callable`. I'll take a look at where it's used and see if we can change it.

Reply:
For context: We already have non-backward compatible changes in service interfaces.

Reply:
I don't see a problem with using `Supplier` here if that makes more sense in an async context. We can leave the other as is or swap `Callable` with `Supplier` in another PR if you want.

Reply:
I switched the async method only. For the sync one, the lambdas passed as arguments throw checked exceptions, so `Callable` makes sense.